src/test/java/com/google/gcp_variant_transforms/library/VcfParserImplTest.java

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 Google LLC
+
+package com.google.gcp_variant_transforms.library;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import com.google.common.collect.ImmutableList;
+import htsjdk.tribble.TribbleException;
+import htsjdk.variant.vcf.VCFCodec;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.junit.MockitoJUnitRunner;
+
+/**
+ * Units tests for VcfParserImpl.java
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class VcfParserImplTest {
+
+  @InjectMocks
+  VcfParserImpl vcfParserImpl;
```

Comment:
Now that I read this part of the code, I think I misunderstood what you were trying to do to tests vcfParserImpl. I don't think we should mock VcfParserImpl here, but instead inject it as is. To do that, please look into having a GuiceBerryRule that sets up a testing environment beforehand, and then we inject vcfParser (not impl - the fake guice rule should still correctly identify the appropriate implementation).

`@Inject VcfParser vcfParser;`

`@Rule GuiceBerryRule guiceBerryRule = new GuiceBerryRule(SomeTestEnvironment.class);`

you'll have to create SomeTestEnvironment class (a guice module, extending AbstractModule) which will act as the default guice module, either for this specific test (in which case we would add it somewhere here at the bottom), or for multiple future modules (probably this is what we should shoot for - if we can reuse the same environment in multiple tests, it would be splendid).

Reply:
The previous testing method is mocking a VcfParserImpl to test all functions. And I have changed by using GuiceBerry to inject VcfParser. And created "TestEnv.java" class that extends the AbstractModule and bind the VcfParser to its Impl.  
And the updated testing injects the VcfParser and tested all the method.  
Resolved.